src/driver/alloy.rs

```diff
@@ -0,0 +1,223 @@
+                        }
+                        ExecutionResult::Success { .. } => {
+                            trevm = res.accept_state();
+                        }
+                    },
+                }
+            }
+
+            // return the final state
+            Ok(trevm)
+        });
+
+        match run_result {
+            Ok(trevm) => Ok(trevm),
+            Err(e) => Err(e),
+        }
+    }
+
+    fn post_bundle<Db: revm::Database + revm::DatabaseCommit>(
+        &mut self,
+        _trevm: &crate::EvmNeedsTx<'_, Ext, Db>,
+    ) -> Result<(), Self::Error<Db>> {
+        Ok(())
+    }
+}
+
+impl<Ext> BundleDriver<Ext> for EthSendBundle {
+    type Error<Db: revm::Database> = BundleError<Db>;
+
+    fn run_bundle<'a, Db: revm::Database + revm::DatabaseCommit>(
+        &mut self,
+        trevm: crate::EvmNeedsTx<'a, Ext, Db>,
+    ) -> DriveBundleResult<'a, Ext, Db, Self> {
+        // 1. Check if the block we're in is valid for this bundle. Both must match
+        if trevm.inner().block().number.to::<u64>() != self.block_number {
+            return Err(trevm.errored(BundleError::BlockNumberMismatch));
+        }
+
+        // 2. Check for start timestamp range validity
+        if let Some(min_timestamp) = self.min_timestamp {
+            if trevm.inner().block().timestamp.to::<u64>() < min_timestamp {
+                return Err(trevm.errored(BundleError::TimestampOutOfRange));
+            }
+        }
+
+        // 3. Check for end timestamp range validity
+        if let Some(max_timestamp) = self.max_timestamp {
+            if trevm.inner().block().timestamp.to::<u64>() > max_timestamp {
+                return Err(trevm.errored(BundleError::TimestampOutOfRange));
+            }
+        }
+
+        let mut t = trevm;
+
+        for raw_tx in self.txs.clone().into_iter() {
+            // Decode the transaction
+            let tx = TxEnvelope::decode_2718(&mut raw_tx.to_vec().as_slice());
+
+            let tx = match tx {
+                Ok(tx) => tx,
+                Err(e) => return Err(t.errored(BundleError::TransactionDecodingError(e))),
+            };
+
+            // Run the transaction
+            let run_result = match t.run_tx(&tx) {
+                Ok(res) => res,
+                Err(e) => return Err(e.map_err(|e| BundleError::EVMError { inner: e })),
+            };
+
+            // Accept the state if the transaction was successful and the bundle did not revert or halt
+            let trevm = match run_result.result() {
+                ExecutionResult::Success { .. } => run_result.accept_state(),
+                ExecutionResult::Revert { .. } | ExecutionResult::Halt { .. } => {
+                    // If the transaction reverted but it is contained in the set of transactions allowed to revert,
+                    // then we discard the state and move on.
+                    if self.reverting_tx_hashes.contains(tx.tx_hash()) {
+                        run_result.reject()
```

Comment:
can we find docs that indicate that allowed reverts are dropped instead of included?

Reply:
good catch, seeing flashbots's rbuilder, they do keep the reverting tx state instead of dropping it if included: https://github.com/flashbots/rbuilder/blob/52fea312e5d8be1f1405c52d1fd207ecee2d14b1/crates/rbuilder/src/building/order_commit.rs#L546-L558